### ISSUE TITLE
[ReactiveElasticsearchTemplate] port defaults to 9200 even when it's not wanted

### ISSUE THREAD

**Issue by bilak** [spring-projects/spring-data-elasticsearch#1683]
Hello,
Today I was playing with this project. I was testing standard template where everything seems to work and also reactive where I had a problem like described on [this](https://stackoverflow.com/questions/64679214/spring-boot-elastic-search-configuration-without-port) SO question. The problem is that we are not using port at all, or at least our url to the elastic doesn't have the port and it's working with standard client. 
When I test the same url in reactive template then I had an error where the port 9200 was appended to the end. 

Can somebody please clarify what is the configuration for `ReactiveElasticsearchTemplate` when the url doesn't contain port?

Thanks

**Comment by sothawo** [comment]
Are you using https? Then you should use `connectedTo("host:443")` and add `usingSsl()` to the configuration.

**Comment by bilak** [comment]
yes that's the https url. Is this somewhere documented please?

**Comment by sothawo** [comment]
Look at the Javadoc: https://github.com/spring-projects/spring-data-elasticsearch/blob/4.2.0-M2/src/main/java/org/springframework/data/elasticsearch/client/ClientConfiguration.java#L191-L197 and https://github.com/spring-projects/spring-data-elasticsearch/blob/4.2.0-M2/src/main/java/org/springframework/data/elasticsearch/client/ClientConfiguration.java#L235-L242

**Comment by bilak** [comment]
That's pretty deep in code. Maybe it would be good to mention this also somewhere in [ref doc](https://docs.spring.io/spring-data/elasticsearch/docs/current/reference/html/#reference) so anybody else will be able to configure that easily in case of issue or when they see default port `9200`. 

But anyway thanks for clarification.

**Comment by sothawo** [comment]
Well the API Doc is not deep in code. It is the primary reference for a library. And any modern IDE will load the API documentation for a library and show it in a tooltip or panel. Besides that, the examples from the reference doc show that the parameters have the form "host:port"

**Comment by bilak** [comment]
@sothawo sorry for bothering, but wasn't there plan to reuse the configuration uris `spring.elasticsearch.rest.uris` for autoconfiguration of reactive rest template? It shouldn't be that hard to parse those informations from URL.

**Comment by sothawo** [comment]
Spring Data Elasticsearch does not use - as well as the other Spring Data modules - configuration entries.

These configuration entries and the whole autoconfiguration stuff is part of Spring Boot. Spring Boot uses these entries to configure Spring Data Elasticsearch, so you would probably need to address this issue with the boot team.
